Makefile

```diff
@@ -0,0 +1,25 @@
+# Copyright 2025 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Author: Philip Wiese <wiesep@iis.ee.ethz.ch>
+
+CLANG_FORMAT_EXECUTABLE ?= clang-format
+
+help:
+	@echo "Usage: make <target>"
+	@echo ""
+	@echo "Available Targets:"
+	@echo " - format: Format all code"
+	@echo " - setup: Install dependencies"
+
+setup:
+	@echo "Installing dependencies..."
+	@pip install -r requirements.txt
```

Comment:
I would personally not do this, as this install process is dependent on the user's environment - I'd go for documenting the installation process in the README, instead.